### PR TITLE
Oracle Based Funding Manager contract structure defined

### PR DESCRIPTION
**Pull request by vendrell46**
This is a WIP, so I've created this PR so that we can review:
- Folder structure
- Contract Inheritance
- Imports
- Naming convention
- Functions proposed for `IOraclePrice_v1` interface (there might be more, but for now it'd be good to know if those added currently are valid)
- Functions proposed for `IERC20Issuance_blacklist_v1` interface

There hasn't been any actual contract implementation done yet, so please ignore function/constructor's implementation.

### CODE REVIEW

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { IERC20PaymentClientBase_v1 } from "@lm/interfaces/IERC20PaymentClientBase_v1.sol";
+
+/**
+ * @title   Oracle Price Funding Manager Interface
+ * @notice  Manages token operations using oracle-based pricing
+ * @dev     Combines funding management and payment client capabilities
+ * @author  Zealynx Security
+ */
+interface IFM_PC_Oracle_Redeeming_v1 is 
+    IERC20PaymentClientBase_v1
```
Comment (FHieser) [inline src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol]: As discussed on slack IFM_PC_Oracle_Redeeming_v1 needs to inherit the IFundingmanager interface, the IERC20PaymentClientBase_v1 and the RedeemingInterface. 
So the only "original" part of the Interface would be the Oracle functionality that should be added here I think, as that is the only new stuff that the other interfaces dont provide

File: src/modules/fundingManager/oracle/OraclePrice_v1.sol [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import { IOraclePrice_v1 } from "./interfaces/IOraclePrice_v1.sol";
+
+/**
+ * @title   Oracle Price Implementation
+ * @notice  Implements oracle price feed functionality
+ * @dev     Provides price calculations for the Funding Manager
+ * @author  Zealynx Security
+ */
+contract OraclePrice_v1 is IOraclePrice_v1 {
```
Comment (FHieser) [inline src/modules/fundingManager/oracle/OraclePrice_v1.sol:12]: Interested for your opinion:
With the explanation I gave in regards of the functionality of this contract would you say that the name fits? If you could rename it what would you choose here?

**Review by FHieser — CHANGES_REQUESTED** [review]
The only thing is the Interface adaption that is still needed. 
Apart from that looks good to me :+1:

File: src/modules/fundingManager/oracle/OraclePrice_v1.sol (hunk 2 above)
Comment (vendrell46) [reply src/modules/fundingManager/oracle/OraclePrice_v1.sol:12]: I'm still not 100% what exactly will this contract end up doing, so hard to guess now. Probably we can leave it as it is and see if we stick to it once we have the Oracle thing clear.

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol (hunk 1 above)
Comment (vendrell46) [reply src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol]: done

Comment (FHieser) [reply src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol]: You are still missing the IRedeemingBondingCurveBase_v1 as inheritance here

**Review by FHieser — CHANGES_REQUESTED** [review]

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol (hunk 1 above)
Comment (vendrell46) [reply src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol]: done, thanks

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * @title   Oracle Price Funding Manager Interface
+ * @notice  Manages token operations using oracle-based pricing
+ * @dev     Combines funding management and payment client capabilities
+ * @author  Zealynx Security
+ */
+interface IFM_PC_Oracle_Redeeming_v1 is
+    IFundingManager_v1,
+    IERC20PaymentClientBase_v1,
+    IRedeemingBondingCurveBase_v1
```
Comment (Zitzak) [inline src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol:17]: Why are we inheriting also the redeeming base interface here? Just curious as I wasn't expecting it

Comment (Zitzak) [reply src/modules/fundingManager/oracle/interfaces/IFM_PC_Oracle_Redeeming_v1.sol:17]: Ahh just got my question answered by seeing the new update in the code of conduct. Consider this comment resolved

**Review by Zitzak — APPROVED** [review]
LGTM, great work!

**Comment by Zitzak** [comment]
The purpose of this PR has been to align on the inheritance setup. @FHieser and I approved the setup (marked by my approval)

However the PR will be closed to comply with the agreement that only PR that pass the CI will be merged into the feature branch. A new PR will be opened soon

**Pull request closed without merging**